dispersy.py

```diff
@@ -1711,11 +1787,12 @@ def store_update_forward(self, possibly_messages, store, update, forward):
 
         store = store and isinstance(messages[0].meta.distribution, SyncDistribution)
         if store:
-            self._store(messages)
+            yield self._store(messages)
 
         if update:
-            if self._update(possibly_messages) == False:
-                return False
+            update_res = yield self._update(possibly_messages)
+            if update_res == False:
```

Comment:
I'd prefer to make minimal changes. Maybe this used to return something else in the past (or will in the future?) and then this check may pass where it should fail